lib/Recur/RRuleIterator.php

```diff
@@ -468,6 +468,13 @@ protected function nextMonthly()
             // beginning.
             $currentDayOfMonth = 0;
 
+            // For some reason the "until" parameter was not being used here,
+            // that's why the workaround of the 10000 year bug was needed at all
+            // let's stop it before the "until" parameter date
+            if ($this->until && $this->currentDate->getTimestamp() >= $this->until->getTimestamp()) {
+                return;
```

Comment:
Codecov reports that there is no test that gets inside this `if` statement,
So you could add a test case that has `until` set. That will verify that this `if` statement can evaluate `true` and return appropriately.

Reply:
Done.